### PR TITLE
Validate chunk size to avoid align crash and add align skip flag

### PR DESCRIPTION
**Pull request by Shivansps**
This fixes the INF_Antaeus.pof crash of Inferno: Alliance due to unknown data found on the model BSP Data chain.

Explanation: 
While parsing the bsp data, the alignment code reads every chunk of the chain one by one, on this particular model on one of the chains at the final 8 bytes there is invalid data that looks like corrupted or trash data, this has an invalid chunk type and a negative chunk size, causing the bsp_pointer to go out of bounds.

Fix:
A check is added to make sure the chunk size value is not negative or has a value that may cause the bsp pointer to go over the max bsp_size.

A troubleshoot flag is also added to disable this process.

### CODE REVIEW

File: code/model/modelread.cpp [hunk 1]
```diff
@@ -6079,6 +6086,14 @@ uint align_bsp_data(ubyte* bsp_in, ubyte* bsp_out, uint bsp_size)
 			memcpy(&bsp_chunk_size, bsp_in + 4, 4);
 		}
 
+		//Chunk size validation, if fails change it to copy the remaining data in chain
+		auto max_size = end - bsp_in;
+		if (bsp_chunk_size < 0 || bsp_chunk_size > max_size) {
+			
+			mprintf(("Invalid BSP Chunk size detected during BSP data align: Chunk Type: %d, Chunk Size: %d, Max Size: %d\n", bsp_chunk_type, bsp_chunk_size, static_cast<uint>(max_size)));
```
Comment (Baezon) [inline code/model/modelread.cpp]: This should be a warning, it's a fairly serious problem the modder should know about.

**Review by Baezon — CHANGES_REQUESTED** [review]

File: code/model/modelread.cpp (hunk 1 above)
Comment (Shivansps) [reply code/model/modelread.cpp]: changed to a warning.

**Review by Baezon — APPROVED** [review]

File: code/model/modelread.cpp [hunk 2]
```diff
@@ -6079,6 +6086,13 @@ uint align_bsp_data(ubyte* bsp_in, ubyte* bsp_out, uint bsp_size)
 			memcpy(&bsp_chunk_size, bsp_in + 4, 4);
 		}
 
+		//Chunk size validation, if fails change it to copy the remaining data in chain
+		auto max_size = end - bsp_in;
+		if (bsp_chunk_size < 0 || bsp_chunk_size > max_size) {
```
Comment (Goober5000) [inline code/model/modelread.cpp:6091]: Methinks neither the change to `int` nor the check for <0 is needed.  A negative number, interpreted as an unsigned value, will always be larger than the largest possible signed value.

File: code/cmdline/cmdline.cpp [hunk 3]
```diff
@@ -2021,6 +2024,10 @@ bool SetCmdlineParams()
 	{
 		Cmdline_lang = lang_arg.str();
 	}
+
+	if (no_bsp_align_arg.found()) {
+		Cmdline_no_bsp_align = 1;
```
Comment (Goober5000) [inline code/cmdline/cmdline.cpp:2029]: Since this is a boolean, it should be assigned `true`, not `1`.

**Review by Goober5000 — CHANGES_REQUESTED** [review]

File: code/model/modelread.cpp (hunk 2 above)
Comment (Shivansps) [reply code/model/modelread.cpp]: you were right about that, it didnt even crossed my mind at the time. I changed it to use uint now.

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**